src/daemon/processing/local_save.rs

```diff
@@ -0,0 +1,69 @@
+use anyhow::Result;
+
+use crate::{
+    daemon::storage::{
+        entities::UsageRecordEntity,
+        record_event::RecordEvent,
+        record_storage::{RecordFileHandle, RecordStorage},
+    },
+    utils::clock::Clock,
+};
+
+use super::module::EventProcessor;
+
+/// Represents saving module. Saving module main goal is to bridge
+/// [ProcessingModule](super::ProcessingModule) and [RecordStorage]
+/// In the future it might also combine multiple savers, like color storage + record storage.
+pub struct LocalSaver<R: RecordStorage> {
```

Comment:
I do not see drop implementation here but it is desired. The thing with futures and overall when any structure holding a file handle is that normal flow might be disrupted. With future there are possibilities when it might be just abandoned by runtime. Such situations can be handled with a drop so that you can flush all buffers and close file handle properly